relint.py

```diff
@@ -79,28 +126,44 @@ def main():
 
     exit_code = 0
 
-    for filename, test, match in matches:
-        exit_code = test.error if exit_code == 0 else exit_code
-        if filename != _filename:
-            _filename = filename
-            lines = match.string.splitlines()
-
-        start_line_no = match.string[:match.start()].count('\n')
-        end_line_no = match.string[:match.end()].count('\n')
-        output_format = "{filename}:{line_no} {test.name}"
-        print(output_format.format(
-            filename=filename, line_no=start_line_no + 1, test=test,
-        ))
-        if test.hint:
-            print("Hint:", test.hint)
-        match_lines = (
-            "{line_no}>    {code_line}".format(
-                line_no=no + start_line_no + 1,
-                code_line=line,
+    if diff:
```

Comment:
that's a bit too much copy paste code here ;) Can you try to not duplicate the print statements?